src/main/java/com/fleotadezuta/youthprogrammanager/service/ChildService.java

```diff
@@ -23,23 +28,41 @@ public class ChildService {
     private final ChildRepository childRepository;
     private final ChildMapper childMapper;
 
-    public Flux<ChildDto> getAllChildren() {
-        return childRepository.findAll()
-                .map(childMapper::fromChildDocumentToChildDto);
+    public Flux<ChildDto> getAllChildren(UserDetails userDetails) {
+        if (userDetails.getUserType().equals("ADMIN")) {
+            return childRepository.findAll()
+                    .map(childMapper::fromChildDocumentToChildDto);
+        } else {
+
+            return childRepository.findChildDocumentsByRelativeParents_Id(userDetails.getUserId())
+                    .map(childMapper::fromChildDocumentToChildDto);
+        }
     }
 
-    public Mono<ChildDto> deleteChild(String id) {
+    public Mono<ChildDto> deleteChild(UserDetails userDetails, String id) {
```

Comment:
Gyereket törölni csak adminnak kellene tudnia, így ebbe a metódusba szerintem nincs is szükség rá, hogy bármilyen módon ellenőrizve legyen a userDetails (mivel a controller-ben szerepel a PreAuthorized annotáció).